pkg/master/controller.go

```diff
@@ -155,15 +174,14 @@ func (c *Controller) CreateMasterServiceIfNeeded(serviceName string, serviceIP n
 			Labels:    map[string]string{"provider": "kubernetes", "component": "apiserver"},
 		},
 		Spec: api.ServiceSpec{
-			Ports: []api.ServicePort{{Port: servicePort, Protocol: api.ProtocolTCP, TargetPort: util.NewIntOrStringFromInt(servicePort)}},
+			Ports: ports,
 			// maintained by this code, not by the pod selector
 			Selector:        nil,
 			ClusterIP:       serviceIP.String(),
```

Comment:
Yup I think we are all in agreement ... The only question is : 
1. Does this feature make sense as a standalone addition to the toolbox of options available for  kube deployments ?  
2. I think "yes" (correct me if wrong)... But I think @thockin is suggesting at the very-least, that I **clarify** the context behind the hacky "launch kubelets via nodeport apiserver" example, which can confuse people into thinking that ClusterIP is not sufficient for internal loadbalanced access.